fix(nuxi): pass through exit code from test errors

piscis opened the pull request:
### 🔗 Linked issue

resolves #18958 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I noticed that my CI will always pass even if I have errors in my nuxt test scripts. Digging through the code of nuxi test command and vitest/node I discovered that errors are displayed but nuxi test will not propagate the test outcome and always exit with exitCode 0  which makes CI pass. 

This should resolve #18958 

Reproduction: https://stackblitz.com/edit/nuxt-starter-el5s4f

Example of a failed test but CI passing: https://github.com/piscis/nuxt-3-nuxi-fails-ci/actions/runs/4151622464/jobs/7182035314#step:6:52

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

piscis committed:
Update cli.ts

Propagate exitCode of sub processes like vitest-node to make CI work with nuxi test

piscis committed:
Fix linter error

manniL commented on packages/nuxi/src/cli.ts:
```diff
@@ -74,7 +74,11 @@ export function main () {
         process.exit(1)
       } else if (result !== 'wait') {
-        process.exit(0)
+        if (process.exitCode && process.exitCode >= 1) {
+          process.exit(process.exitCode)
+        } else {
+          process.exit(0)
+        }
```
Is there any reason not to always all `.exit` with the exitCode?

```suggestion
        process.exit(process.exitCode)
```

piscis replied:
I thing setting the exit code can be emitted all together because the default is 0 and then other processes can overwritten by process.exitCode. That’s what vitest tries todo.

piscis replied:
From the manual: Specifying a code to [process.exit(code)](https://nodejs.org/api/process.html#processexitcode) will override any previous setting of process.exitCode.

piscis committed:
Omit exit code 

Omits the exit code all together if process.exitCode is defined then this is used to exit otherwise it’s 0 by default

piscis committed:
Merge branch 'main' into patch-1

danielroe committed:
Merge branch 'main' into patch-1
